src/rgw/rgw_rest_s3.cc

```diff
@@ -1589,6 +1589,8 @@ int RGWPostObj_ObjStore_S3::read_form_part_header(struct post_form_part *part,
       break;
 
     r = read_line(bl, chunk_size, &reached_boundary, done);
+    if (r < 0)
+        return r;
```

Comment:
@wangzhengyong looks good. It looks like we probably also need to add to check `if (*done)` like above. Maybe change the loop to `while (!*done)`?